server/dashboards/preconfigured/bot-framework.js

```diff
@@ -406,12 +406,12 @@ return {
               channel: (val) => val || "unknown",
               channel_count: (val) => val || 0
             },
-            calculated: (filterChannels) => {
+            calculated: (filterChannels, dependencies, prevState) => {
               const filters = filterChannels.map((x) => x.channel);
-              let { selectedValues } = filterChannels;
-              if (selectedValues === undefined) {
-                selectedValues = [];
-              }
+              let selectedValues = [];
+              if (prevState['channels-selected'] !== undefined) {
```

Comment:
I understand the necessity, but can we possibly hide this logic as a flag of the filter, rather than complicating the data source?